integration_tests/test_dbapi.py

```diff
@@ -87,24 +91,141 @@ def test_select_query_result_iteration_statement_params(presto_connection):
         params=(3,)  # expecting all the rows with id >= 3
     )
 
-def test_select_query_result_string_statement_params(presto_connection):
-    """
-    Tests string parameters in prepared statements
-    """
+
+def test_none_query_param(presto_connection):
+    cur = presto_connection.cursor()
+    cur.execute("SELECT ?", params=(None,))
+    rows = cur.fetchall()
+
+    assert rows[0][0] == None
+
+
+def test_string_query_param(presto_connection):
     cur = presto_connection.cursor()
-    # test a simple string parameter and
-    # a parameter with a quote in it
-    cur.execute(
-        "select ?",
-        params=("six'",),
-    )
 
+    cur.execute("SELECT ?", params=("six'",))
     rows = cur.fetchall()
 
-    assert len(rows) == 1
     assert rows[0][0] == "six'"
 
 
+def test_datetime_query_param(presto_connection):
+    cur = presto_connection.cursor()
+
+    cur.execute(
+            "SELECT ?", 
+            params=(datetime(2020, 1, 1, 0, 0, 0),)
+            )
+    rows = cur.fetchall()
+
+    assert rows[0][0] == "2020-01-01 00:00:00.000"
+
+    cur.execute(
+            "SELECT ?", 
+            params=(datetime(2020, 1, 1, 0, 0, 0, tzinfo=pytz.utc),)
+            )
+    rows = cur.fetchall()
+
+    assert rows[0][0] == "2020-01-01 00:00:00.000 UTC"
+    assert cur.description[0][1] == "timestamp with time zone"
+
+
+def test_array_query_param(presto_connection):
+    cur = presto_connection.cursor()
+
+    cur.execute("SELECT ?", params=([1, 2, 3],))
+    rows = cur.fetchall()
+
+    assert rows[0][0] == [1, 2, 3]
+
+    cur.execute(
+            "SELECT ?", 
+            params=([[1, 2, 3],[4,5,6]],))
+    rows = cur.fetchall()
+
+    assert rows[0][0] == [[1, 2, 3],[4,5,6]]
+
+    cur.execute("SELECT TYPEOF(?)", params=([1, 2, 3],))
+    rows = cur.fetchall()
+
+    assert rows[0][0] == "array(integer)"
+
+
+def test_dict_query_param(presto_connection):
+    cur = presto_connection.cursor()
+
+    cur.execute("SELECT ?", params=({"foo": "bar"},))
+    rows = cur.fetchall()
+
+    assert rows[0][0] == {"foo": "bar"}
+
+    cur.execute("SELECT TYPEOF(?)", params=({"foo": "bar"},))
+    rows = cur.fetchall()
+
+    assert rows[0][0] == "map(varchar(3), varchar(3))"
+
+
+def test_boolean_query_param(presto_connection):
+    cur = presto_connection.cursor()
+
+    cur.execute("SELECT ?", params=(True,))
+    rows = cur.fetchall()
+
+    assert rows[0][0] == True
+
+    cur.execute("SELECT ?", params=(False,))
+    rows = cur.fetchall()
+
+    assert rows[0][0] == False
+
+
+def test_float_query_param(presto_connection):
+    cur = presto_connection.cursor()
+    cur.execute("SELECT ?", params=(1.1,))
+    rows = cur.fetchall()
+
+    assert rows[0][0] == 1.1
```

Comment:
let's have `cur.description`  assertion for floats too 
(eg to "document" floats end up as being `double` in Presto)